documentation/book/cluster-operator.adoc

```diff
@@ -673,6 +673,116 @@ INFO:: The Cluster Operator doesn't validate the provided configuration. When in
 Kafka Connect cluster might not start or might become unstable. In such cases, the configuration in the `connect-config` field
 should be fixed and the Cluster Operator will roll out the new configuration to all Kafka Connect instances.
 
+===== Kafka Connect S2I deployment
+
+When using {ProductName} together with an {OpenShiftName} cluster, Kafka Connect can be deployed with support for https://docs.openshift.org/3.9/dev_guide/builds/index.html[{OpenShiftName} Builds] and https://docs.openshift.org/3.9/creating_images/s2i.html#creating-images-s2i[Source-to-Image (S2I)].
+To activate the S2I deployment, the `strimzi.io/type` label in the ConfigMap should be set to `kafka-connect-s2i`.
+The following is a full example of a ConfigMap for a Kafka Connect S2I cluster.
+
+.Example Kafka Connect S2I cluster ConfigMap
+[source,yaml,options="nowrap",subs="attributes"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-connect-cluster
+  labels:
+    strimzi.io/kind: cluster
+    strimzi.io/type: kafka-connect-s2i
+data:
+  nodes: "1"
+  image: "{DockerKafkaConnectS2I}"
+  healthcheck-delay: "60"
+  healthcheck-timeout: "5"
+  connect-config: |-
+    {
+      "bootstrap.servers": "my-cluster-kafka:9092"
+    }
+----
+
+The S2I deployment is very similar to the regular Kafka Connect deployment.
+Compared to the regular deployment, the Cluster Operator will create the following additional resources:
+
+* [connect-cluster-name]-connect-source ImageStream which is used as the base image for the newly-built Docker images
+* [connect-cluster-name]-connect BuildConfig which is responsible for building the new Kafka Connect Docker images
+* [connect-cluster-name]-connect ImageStream where the newly built Docker images will be pushed
+* [connect-cluster-name]-connect DeploymentConfig which is in charge of creating the Kafka Connect worker node pods
+* [connect-cluster-name]-connect Service which exposes the REST interface for managing the Kafka Connect cluster
+
+The Kafka Connect S2I deployment supports the same options as the regular Kafka Connect deployment.
+A list of supported options can be found in the <<kafka_connect_config_map_details>> section.
+The `image` option specifies the Docker image which will be used as the _source image_ - the base image for the newly build Docker image.
+The default value of the `image` option is determined by the value of the `<<STRIMZI_DEFAULT_KAFKA_CONNECT_S2I_IMAGE,STRIMZI_DEFAULT_KAFKA_CONNECT_S2I_IMAGE>>` environment variable of the Cluster Operator.
+All other options have the same meaning as for the regular deployment.
+
+Once the Kafka Connect S2I cluster is deployed, new plugins can be added by starting a new {OpenShiftName} build.
+The new build can be started using the `oc` command line tool.
```

Comment:
I think you can drop this sentence: It doesn't really add anything that isn't covered by line 754, and having it here is too soon, since they first have to add assemble the directory of plugins.